src/Compute/Compute/VirtualMachine/Config/SetAzureVMOperatingSystemCommand.cs

```diff
@@ -306,11 +306,20 @@ public override void ExecuteCmdlet()
         {
             if (this.VM.OSProfile == null)
             {
+                if (this.Credential == null)
+                {
+                    throw new Exception("A credential is required to set the VM operating system when the VM does not have an OS Profile set. Please provide a credential to the -Credential parameter.");
+                }
+                if (this.ComputerName == null)
+                {
+                    WriteInformation("A computer name will be required when creating a virtual machine. We suggest you provide a computer name in the -ComputerName parameter.", new string[] { "PSHOST" });
+                }
+
                 this.VM.OSProfile = new OSProfile
                 {
-                    ComputerName = this.ComputerName,
-                    AdminUsername = this.Credential.UserName,
-                    AdminPassword = ConversionUtilities.SecureStringToString(this.Credential.Password),
+                    ComputerName = string.IsNullOrWhiteSpace(this.ComputerName) ? null : this.ComputerName,
+                    AdminUsername = this.Credential.UserName == null ? null : this.Credential.UserName,
```

Comment:
this should take care of the null reference error, it UserName and password will pass 'null' if credentials not provided.

Do we want to defer the error message for needing credentials to the service side?